### PR TITLE
Bugfix/ees 857 prevent mutiple submit

### PR DESCRIPTION
**Pull request by shakes90**

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx [hunk 1]
```diff
@@ -163,13 +163,18 @@ const ReleaseDataUploadsSection = ({ publicationId, releaseId }: Props) => {
...
-    setIsUploading(false);
-    await resetPage(actions);
+    await editReleaseDataService
+      .uploadDataFiles(releaseId, {
+        subjectTitle: values.subjectTitle,
+        dataFile: values.dataFile as File,
+        metadataFile: values.metadataFile as File,
+      })
+      .then(() => {
+        resetPage(actions);
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx]: Shouldn't have to change this all to use promise syntax. Wrapping the entire thing with a try/finally would have sufficed. Async/await and promises are all interchangeable, but I personally prefer async/await in the majority of cases.

If you do want to continue using promise syntax, then we should at least return the `resetPage` here as it is an async task itself and we wouldn't want to `setIsUploading(false)` until after the page has finished resetting i.e.

```typescript
.then(() => resetPage(actions));
```

File: src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx [hunk 2]
```diff
@@ -180,9 +185,11 @@ const ReleaseDataUploadsSection = ({ publicationId, releaseId }: Props) => {
     setDeleteDataFile(undefined);
     await editReleaseDataService
       .deleteDataFiles(releaseId, (deleteDataFile as DeleteDataFile).file)
+      .then(() => {
+        resetPage(form);
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx]: Same as above

File: src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseFileUploadsSection.tsx [hunk 3]
```diff
@@ -85,12 +84,17 @@ const ReleaseFileUploadsSection = ({ publicationId, releaseId }: Props) => {
...
-    });
-    setIsUploading(false);
-    await resetPage(actions);
+    await editReleaseDataService
+      .uploadAncillaryFile(releaseId, {
+        name: values.name,
+        file: values.file as File,
+      })
+      .then(() => {
+        resetPage(actions);
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseFileUploadsSection.tsx]: Same as comments as in `ReleaseDataUploadsSection`

File: src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx (hunk 1 above)
Comment (shakes90) [reply src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx]: Have changed the logic slightly to ensure to clear the spinner before resetting the form which is the only way with the current error handling logic that works.

Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx]: Cool, not going to delve too deep into it, but it seems like this new change should also be applied to `RleeaseDataUploadsSection` too?

**Pull request merged**